src/adapters/http/download/router.ts

```diff
@@ -1,9 +1,23 @@
 import path from 'path';
 import express from 'express';
+import { constant, pipe } from 'fp-ts/lib/function';
+import * as TE from 'fp-ts/TaskEither';
+import { SystemEnv } from '../../../useCases/SystemEnv';
+import { makeDownloadRecord } from '../../../domain/DownloadRecord';
+import { persistRecord } from '../../../useCases/PersistRecord';
 
-export const makeDownloadDocumentRouter = (): express.Router => {
+const persistDownloadRecord =
+  (env: SystemEnv) => (req: express.Request, _res: express.Response, next: express.NextFunction) =>
```

Comment:
In this case, I think we should define the return type we are expecting. Just because the `app.use` method is too open

Reply:
Yep, I've made some changes